Source/Core/DiscIO/NANDImporter.cpp

```diff
@@ -90,32 +91,34 @@ bool NANDImporter::ReadNANDBin(const std::string& path_to_bin,
   return file.ReadBytes(m_nand_keys.data(), NAND_KEYS_SIZE);
 }
 
-void NANDImporter::FindSuperblock()
+bool NANDImporter::FindSuperblock()
 {
   constexpr size_t NAND_SUPERBLOCK_START = 0x1fc00000;
-  constexpr size_t NAND_SUPERBLOCK_SIZE = 0x40000;
 
-  size_t superblock = 0;
-  u32 newest_version = 0;
-  for (size_t pos = NAND_SUPERBLOCK_START; pos < NAND_SIZE; pos += NAND_SUPERBLOCK_SIZE)
+  // There are 16 superblocks, choose the highest/newest version
+  for (int i = 0; i < 16; i++)
   {
-    if (!memcmp(m_nand.data() + pos, "SFFS", 4))
-    {
-      const u32 version = Common::swap32(&m_nand[pos + 4]);
-      INFO_LOG_FMT(DISCIO, "Found superblock at {:#x} with version {:#x}", pos, version);
-      if (superblock == 0 || version > newest_version)
-      {
-        superblock = pos;
-        newest_version = version;
-      }
-    }
+    auto superblock = std::make_unique<NANDSuperblock>();
+    std::memcpy(superblock.get(), &m_nand[NAND_SUPERBLOCK_START + i * sizeof(NANDSuperblock)],
+                sizeof(NANDSuperblock));
+
+    if (std::memcmp(superblock->magic, "SFFS", 4) != 0)
+      continue;
```

Comment:
Should a warning be logged in this case, or is this something that happens with normal NAND dumps?